pkg/operator/workloadcontroller/workload_controller_openshiftapiserver.go

```diff
@@ -216,22 +215,31 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
 		c.versionRecorder.SetVersion("openshift-apiserver", operandVersion)
 
 	}
-	if !equality.Semantic.DeepEqual(operatorConfig.Status, originalOperatorConfig.Status) {
-		if _, err := c.operatorConfigClient.OpenShiftAPIServers().UpdateStatus(operatorConfig); err != nil {
-			return false, err
-		}
+
+	currOperatorStatus, _, err := v1helpers.UpdateStatus(
```

Comment:
if UpdateStatus() determines new status == old status, won't currOperatorStatus will be returned as nil? If yes, we need to add a check for nil and an early return otherwise expect issues at line 236